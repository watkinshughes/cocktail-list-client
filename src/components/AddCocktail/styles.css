.AddCocktail {
  clear: both;
}

.AddCocktail input[type="text"],
.AddCocktail textarea,
.AddCocktail select {
  -webkit-appearance: none;
  border: 1px solid #ccc;
  padding: 20px 0;
  margin: 10px 0;
  display: block;
  width: 100%;
  text-indent: 1em;
}

.AddCocktail button[type="submit"] {
  -webkit-appearance: none;
  border: 1px solid #ccc;
  background: #333;
  color: #fff;
  padding: 20px 0;
  margin: 10px 0;
  display: block;
  width: 100%;
  font-size: 20px;
  text-indent: 1em;
  cursor: pointer;
}