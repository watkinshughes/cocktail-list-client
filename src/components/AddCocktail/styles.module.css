.form {
  clear: both;
}

.input {
  -webkit-appearance: none;
  border: 1px solid #ccc;
  padding: 20px 0;
  margin: 10px 0;
  display: block;
  width: 100%;
  text-indent: 1em;
}

.button {
  -webkit-appearance: none;
  border: 1px solid #ccc;
  background: #333;
  color: #fff;
  padding: 20px 0;
  margin: 10px 0;
  display: block;
  width: 100%;
  font-size: 20px;
  text-indent: 1em;
  cursor: pointer;
}