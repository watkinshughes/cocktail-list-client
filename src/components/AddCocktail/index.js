import React, { Component } from "react";
import axios from "axios";
import './styles.css'

class AddCocktail extends Component {
  state = {
    name: "",
    glass: "",
    category: "",
    garnish: "",
    preparation: "",
    ingredients: ""
  };

  handleChange = event => {
    this.setState({
      name: event.target.name,
      glass: event.target.name,
      category: event.target.name,
      garnish: event.target.name,
      preparation: event.target.name,
      ingredients: event.target.name
    });
  };

  handleSubmit = event => {
    event.preventDefault();

    axios
      .post(`https://cocktail-list-api.herokuapp.com/cocktails/`, {
        name: this.state.name,
        glass: this.state.glass,
        category: this.state.category,
        garnish: this.state.garnish,
        preparation: this.state.preparation,
        ingredients: this.state.ingredients
      })
      .then(response => {
        console.log(response.data);
      })
      .catch(error => {
        console.log("An error occurred:", error);
      });
  };

  render() {
    return (
      <div>
        <form className="AddCocktail" onSubmit={this.handleSubmit}>
          <label>
            <div className="visually-hidden">Name</div>
            <input type="text" name="name" placeholder="Name" onChange={this.handleChange} />
          </label>
          <label>
            <div className="visually-hidden">Category</div>
              <select name="category" onChange={this.handleChange}>
                <option value="">Select Category</option>
                <option value="Before Dinner Cocktail">Before Dinner Cocktail</option>
                <option value="After Dinner Cocktail">After Dinner Cocktail</option>
                <option value="All Day Cocktail">All Day Cocktail</option>
                <option value="Longdrink">Longdrink</option>
              </select>
          </label>
          <label>
            <div className="visually-hidden">Glass</div>
            <select name="glass" onChange={this.handleChange}>
                <option value="">Select Glass</option>
                <option value="martini">Martini</option>
                <option value="old-fashioned">Old-fashioned</option>
                <option value="collins">Collins</option>
                <option value="highball">Highball</option>
                <option value="champagne">Champagne</option>
                <option value="margarita">Margarita</option>
              </select>
          </label>
          <label>
            <div className="visually-hidden">Garnish</div>
            <input type="text" placeholder="Garnish" name="granish" onChange={this.handleChange} />
          </label>
          <label>
            <div className="visually-hidden">Preparation</div>
            <textarea
              name="preparation"
              placeholder="Preparation"
              onChange={this.handleChange}
            />
          </label>
          <label>
            <div className="visually-hidden">Ingredients</div>
            <textarea
              name="ingredients"
              placeholder="Ingredients"
              onChange={this.handleChange}
            />
          </label>
          <div>
            <button type="submit">Submit</button>
          </div>
        </form>
      </div>
    );
  }
}

export default AddCocktail;
