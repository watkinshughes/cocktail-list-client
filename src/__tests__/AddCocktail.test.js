import React from "react";
import ReactDOM from "react-dom";
import { create } from "react-test-renderer";
import AddCocktail from "../components/AddCocktail";

test("snapsot", () => {
  const c = create(<AddCocktail />);
  expect(c.toJSON()).toMatchSnapshot();
});

it("renders without crashing", () => {
  const div = document.createElement("div");
  ReactDOM.render(<AddCocktail />, div);
  ReactDOM.unmountComponentAtNode(div);
});
