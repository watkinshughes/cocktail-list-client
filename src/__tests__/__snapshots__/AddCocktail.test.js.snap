// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapsot 1`] = `
<form
  className="form"
  onSubmit={[Function]}
>
  <label>
    <span
      className="visually-hidden"
    >
      Name
    </span>
    <input
      name="name"
      onChange={[Function]}
      placeholder="Name"
      type="text"
    />
  </label>
  <label>
    <span
      className="visually-hidden"
    >
      Category
    </span>
    <select
      name="category"
      onChange={[Function]}
    >
      <option
        value=""
      >
        Select Category
      </option>
      <option
        value="Before Dinner Cocktail"
      >
        Before Dinner Cocktail
      </option>
      <option
        value="After Dinner Cocktail"
      >
        After Dinner Cocktail
      </option>
      <option
        value="All Day Cocktail"
      >
        All Day Cocktail
      </option>
      <option
        value="Longdrink"
      >
        Longdrink
      </option>
    </select>
  </label>
  <label>
    <span
      className="visually-hidden"
    >
      Glass
    </span>
    <select
      name="glass"
      onChange={[Function]}
    >
      <option
        value=""
      >
        Select Glass
      </option>
      <option
        value="martini"
      >
        Martini
      </option>
      <option
        value="old-fashioned"
      >
        Old-fashioned
      </option>
      <option
        value="collins"
      >
        Collins
      </option>
      <option
        value="highball"
      >
        Highball
      </option>
      <option
        value="champagne"
      >
        Champagne
      </option>
      <option
        value="margarita"
      >
        Margarita
      </option>
    </select>
  </label>
  <label>
    <span
      className="visually-hidden"
    >
      Garnish
    </span>
    <input
      name="granish"
      onChange={[Function]}
      placeholder="Garnish"
      type="text"
    />
  </label>
  <label>
    <span
      className="visually-hidden"
    >
      Preparation
    </span>
    <textarea
      name="preparation"
      onChange={[Function]}
      placeholder="Preparation"
    />
  </label>
  <label>
    <span
      className="visually-hidden"
    >
      Ingredients
    </span>
    <textarea
      name="ingredients"
      onChange={[Function]}
      placeholder="Ingredients"
    />
  </label>
  <div>
    <button
      className="button"
      type="submit"
    >
      Submit
    </button>
  </div>
</form>
`;
